### DISCUSSION TITLE
Updated UI

### DISCUSSION THREAD

**Discussion by andycb** [andycb/AdventurerClientJS#43] [Development]
Creating a discussion thread to continue discussion an updated UI based on the conversation started on #39.

**Comment by Gaweringo** [comment]
## First mockup of the status page
![Status page with camera](https://user-images.githubusercontent.com/40121865/107365355-f888db00-6adc-11eb-9c0e-88e9acca7b52.png)
This design would show everything that is changing on the main status page. (As [mentioned](https://github.com/andycb/AdventurerClientJS/issues/32#issuecomment-775530597) by @orangevan in #32 the print head position and used filament could also be displayed)

Clicking on the name, or maybe a link next to it could reveal the 'static' information about the printer, like the Version, Serial Number or Model.
![Printer Information](https://user-images.githubusercontent.com/40121865/107393707-6217e200-6afb-11eb-8d80-435aecfab8b5.png)

For printers that don't have a camera built in like the _Adventurer 3 Lite_ the space where the camera would normally be placed could be taken up by the static information.
![Status page without camera](https://user-images.githubusercontent.com/40121865/107394123-cdfa4a80-6afb-11eb-9a9c-44dcbd6819af.png)
I'm not entirely satisfied with this as it leaves a lot of blank space.

And should we add a settings page to disable the camera? With just one setting this page would be very empty, so I'm not sure what we should do here.
Maybe reordering the header would also be a good idea then
![Reordered header](https://user-images.githubusercontent.com/40121865/107395083-ce471580-6afc-11eb-9c0e-26ad410fec91.png)


### With this wider design, the other pages would look like this
![Print File page wide](https://user-images.githubusercontent.com/40121865/107367443-98476880-6adf-11eb-99da-ca64e83de32b.png)
In my opinion, the print page doesn't look too bad like this.

![About page wide](https://user-images.githubusercontent.com/40121865/107367497-a85f4800-6adf-11eb-813c-c09818902079.png)
The text on the about page should definitely be constrained in its width.

![Debug page wide](https://user-images.githubusercontent.com/40121865/107393521-3137ad00-6afb-11eb-966e-db038ea17c93.png)
The same with the debug page.

**Reply by andycb** [reply]
Love this.

Agreed that showing more information in the camera area for more printers with no camera is perhaps not ideal, as it means the app had two different views based on the type of printer its connected to.

Question: The window is resizable, do you have some thoughts to how to handle the window being made smaller? Would the camera feed get smaller / move as the window was made narrower.

**Reply by Gaweringo** [reply]
Yeah, maybe the camera could get smaller, but at some stage, it would probably be best to put it under the Information.
Maybe like this:
#### Smallest the camera should be
![Camera smaller](https://user-images.githubusercontent.com/40121865/107591103-32a3cb00-6c0a-11eb-8e29-a91d4319cf41.png)
#### Then put the camera under the information and center everything
![Camera under Information](https://user-images.githubusercontent.com/40121865/107591124-3d5e6000-6c0a-11eb-9f62-6c69e267e8ea.png)
![Smallest size scrolling](https://user-images.githubusercontent.com/40121865/107591135-4818f500-6c0a-11eb-858b-62ad48bb240a.gif)
And I think it would be smart to set a min-width for the application that still fits what's needed.

### And if another section like settings is added I think there would be a need for an overflow menu
![Overflow menu](https://user-images.githubusercontent.com/40121865/107592033-346e8e00-6c0c-11eb-97e4-28ab3a59ad4d.gif)

**Comment by Gaweringo** [comment]
#26 could be implemented like this
![Status page with printing progress](https://user-images.githubusercontent.com/40121865/107428745-61934180-6b23-11eb-8a58-2eb9f1053721.png)
